desktop/notification/hints.go

```diff
@@ -0,0 +1,204 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package notification
+
+import "fmt"
+
+// WithActionIcons returns a hint asking the server to use action key as icon names.
+//
+// A server that has the "action-icons" capability will attempt to interpret any
+// action key as a named icon. The localized display name will be used to
+// annotate the icon for accessibility purposes. The icon name should be
+// compliant with the Freedesktop.org Icon Naming Specification.
+//
+// Requires server version >= 1.2
+func WithActionIcons() Hint {
+	t := true
+	return Hint{Name: "action-icons", Value: &t}
+}
+
+// Urgency describes the importance of a notification message.
+type Urgency byte
+
+const (
+	// LowUrgency indicates that a notification message is below normal priority.
+	LowUrgency Urgency = 0
+	// NormalUrgency indicates that a notification message has the regular priority.
+	NormalUrgency Urgency = 1
+	// CriticalUrgency indicates that a notification message is above normal priority.
+	CriticalUrgency Urgency = 2
+)
+
+// String implements the Stringer interface.
+func (u Urgency) String() string {
+	switch u {
+	case LowUrgency:
+		return "low"
+	case NormalUrgency:
+		return "normal"
+	case CriticalUrgency:
+		return "critical"
+	default:
+		return fmt.Sprintf("Urgency(%d)", byte(u))
+	}
+}
+
+// WithUrgency returns a hint asking the server to set message urgency.
+//
+// Notification servers may show messages with higher urgency before messages
+// with lower urgency. In addition some urgency levels may not be shown when the
+// user has enabled a do-not-distrub mode.
+func WithUrgency(u Urgency) Hint {
+	return Hint{Name: "urgency", Value: &u}
+}
+
+// Category is a string indicating the category of a notification message.
+type Category string
+
+const (
```

Comment:
We probably won't need most of those but given that we have them now they also won't hurt :)

Reply:
Yeah, I think those don't hurt. I could also fork those off to a separate repository where we have the full thing and we use it from snapd but I wanted to avoid introducing what looks like a dependency as that's always a pain.